Server/src/controllers/getDatasetController.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import { retrieveData } from '../services/getDatasetService';
+import { IDataRequestModel } from "../models/interfaces/DataRequestModelInterface";
+
+export class getDataController {
+    private processedRequest: IDataRequestModel;
+    constructor() {
+    }
+
+    /**
+     * This controller will take a request, verify that it is valid, and if it valid will then process the request,
+     * send it to the getDataService to acquire an array of data sets that match the entered search terms, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForData(request: Request, response: Response) {
+        let validateData = this.validateInputData(request)
+        if (!validateData) {
+            response.status(500).send("Invalid search params entered");

```

Comment:
Error 500 is a server related error, if the params are invalid we should either throw a badRequest error or instead return with error 400 instead